src/Utils/Utils.php

```diff
@@ -272,9 +272,9 @@ public static function groupBy($traversable, callable $keyFn)
     }
 
     /**
-     * @param mixed[]|Traversable $traversable
+     * @param array<mixed, mixed>|Traversable $traversable
```

Comment:
Array cannot have `mixed` in keys. Also we can add documentation for Traversable type or change it to `iterable` because `array|Traversable = iterable`

Reply:
What would be the type for the key then?

Reply:
For arrays it can only be `int|string`

Reply:
What about `float` or `bool`?

Reply:
arrays can't have bool or float in keys https://3v4l.org/rQen6 It will always be subset of `int|string`